### ISSUE TITLE
Avoid template variable collision during template rendering in View.php

### ISSUE THREAD

**Issue by qlimax86** [mikecao/flight#90]
i suggest to change, in flight/Template/View.php line 109
this

``` PHP
extract($this->vars);
```

to

``` PHP
extract($this->vars,EXTR_SKIP);
```

so, in case, you skip the override of the variable named 'template'.

Keep up the great work !

**Comment by mikecao** [comment]
I actually prefer the overwrite behavior because it is more explicit. It's easier to manage the naming of your variables than it is to debug why a variable is not set. But it's really not an issue here because `extract` in this case only creates variables within the template scope, not global.

**Comment by qlimax86** [comment]
Of course, extract() does create (and overwrites) the variables in the function local scope.

Because of that, if I'm not wrong, the call

``` PHP
Flight:render('existingview.php',array('template'=>'notexisting.php'));
```

would raise a fatal error, because $template would be overwritten.
And the include function would try to include notexisting.php

**Comment by mikecao** [comment]
I see what you mean. I'll work on a fix.

**Comment by mikecao** [comment]
Fixed in commit 6239ab4d8038ed807fa29e3622db9b1dfb2da181.
